src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<? extends Machine> machineProducer;
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+        } else if (type == Truck.class) {
+            machineProducer = new TruckProducer();
+        } else if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
+        } else {
+            return Collections.emptyList();
+        }
```

Comment:
lets use switch-case here. And it's better to throw some exception in case type do not fit one of the conditions

Reply:
I tried but - not incompatible type in "switch()" parathesis  ...